Foundation/System/Bindings/POSIX/Stropts.hsc

```diff
@@ -0,0 +1,235 @@
+        #{poke struct winsize, ws_ypixel} ptr y
+        
+#elif defined FOUNDATION_SYSTEM_WINDOWS
+newtype Handle = Ptr ()
+
+data SmallRect = SmallRect {
+    left :: CShort,
+    top :: CShort,
+    right :: CShort,
+    bottom :: CShort
+}
+
+instance Storable SmallRect where
+    sizeOf _ = #{size SMALL_RECT}
+    alignment _ = #{alignment SMALL_RECT}
+    peek ptr = do
+        l <- #{peek SMALL_RECT, Left} ptr
+        t <- #{peek SMALL_RECT, Right} ptr
+        r <- #{peek SMALL_RECT, Top} ptr
+        b <- #{peek SMALL_RECT, Bottom} ptr
+        return (SmallRect l t r b)
+    poke ptr (SmallRect l t r b) = do
+        #{poke SMALL_RECT, Left} ptr l
+        #{poke SMALL_RECT, Top} ptr t
+        #{poke SMALL_RECT, Right} ptr r
+        #{poke SMALL_RECT, Bottom} ptr b
+        
+data Coord = Coord {
+    x :: CShort,
+    y :: CShort
+}
+
+instance Storable Coord where
+    sizeOf _ = #{size Coord}
+    alignment _ = #{alignment Coord}
+    peek ptr = do
+        x <- #{peek Coord, X} ptr
+        y <- #{peek Coord, Y} ptr
+        return (Coord x y)
+    poke ptr (Coord x y) = do
+        #{poke Coord, x} ptr x
+        #{poke Coord, x} ptr y
+
+data ConsoleScreenBufferInfo = ConsoleScreenBufferInfo {
+    dwSize :: Coord,
+    dwCursorPosition :: Coord,
+    wAttributes :: CUShort,
+    srWindow :: SmallRect,
+    dwMaximumWindowSize :: Coord,
+}
+
+instance Storable ConsoleScreenBufferInfo where
+    sizeOf _ = #{size ConsoleScreenBufferInfo}
+    alignment _ = #{alignment ConsoleScreenBufferInfo}
+    peek ptr = do
+        s <- #{peek ConsoleScreenBufferInfo, dwSize} ptr
+        c <- #{peek ConsoleScreenBufferInfo, dwCursorPosition} ptr
+        a <- #{peek ConsoleScreenBufferInfo, wAttributes} ptr
+        w <- #{peek ConsoleScreenBufferInfo, srWindow} ptr
+        m <- #{peek ConsoleScreenBufferInfo, dwMaximumWindowSize} ptr
+        return (ConsoleScreenBufferInfo s c a w m)
+    poke ptr (ConsoleScreenBufferInfo s c a w m) = do
+        #{poke ConsoleScreenBufferInfo, dwSize} ptr s
+        #{poke ConsoleScreenBufferInfo, dwCursorPosition} ptr c
+        #{poke ConsoleScreenBufferInfo, wAttributes} ptr a
+        #{poke ConsoleScreenBufferInfo, srWindow} ptr w
+        #{poke ConsoleScreenBufferInfo, dwMaximumWindowSize} ptr m
+    
+invalidHandleValue :: IntPtr
+invalidHandleValue = #{const INVALID_HANDLE_VALUE}
+
+stdInputHandle :: CULong
+stdInputHandle = #{const STD_INPUT_HANDLE}
+#endif
+-- defined FOUNDATION_SYSTEM_WINDOWS
+
+
+foreign import ccall "popen" c_popen :: CString -> CString -> IO (Ptr CFile)
+foreign import ccall "pclose" c_pclose :: Ptr CFile -> IO CInt
+foreign import ccall "fscanf_int" c_fscanf_int :: Ptr CFile -> Ptr CInt -> IO CInt
+#ifdef FOUNDATION_SYSTEM_UNIX
+foreign import ccall "ioctl_winsize" c_ioctl_winsize :: CInt -> Ptr Winsize -> IO CInt
+#elif defined FOUNDATION_SYSTEM_WINDOWS
+foreign import ccall "GetStdHandle" c_get_std_handle :: CULong -> IO Handle
```

Comment:
I think this should be stdcall for windows 32 bits here.
also I'm surprise this is not available in `win32` already, I think GetStdHandle is available through `Graphics.Win32.Misc`, but I think it would probably to contribute GetConsoleScreenBufferInfo in `win32` too !